src/main.rs

```diff
@@ -129,21 +143,21 @@ async fn upload_project(
 
     // Create an upload ID and temporary directory.
     let upload_id = Uuid::new_v4();
-    let upload_dir_str = format!("tmp/uploads/{}", upload_id);
-    let upload_dir = Path::new(&upload_dir_str);
+    let tmp_dir = tempdir().map_err(|_| ApiError::Upload(UploadError::CreateTempDir))?;
+    let upload_dir = tmp_dir.path().join(upload_id.to_string());
 
-    fs::create_dir_all(upload_dir).map_err(|_| ApiError::Upload(UploadError::SaveFile))?;
+    fs::create_dir(upload_dir.clone()).map_err(|_| ApiError::Upload(UploadError::SaveFile))?;
```

Comment:
not sure if this is required since we already have a tempdir available

Reply:
It's required since I'm creating a subdir inside of the tmpdir. You could argue that I don't need the subdir but I'll leave it as is for now.